Mathlib/Algebra/GroupWithZero/Units/Basic.lean

```diff
@@ -0,0 +1,370 @@
+
+theorem div_ne_zero (ha : a ≠ 0) (hb : b ≠ 0) : a / b ≠ 0 := by
+  rw [div_eq_mul_inv]
+  exact mul_ne_zero ha (inv_ne_zero hb)
+#align div_ne_zero div_ne_zero
+
+@[simp]
+theorem div_eq_zero_iff : a / b = 0 ↔ a = 0 ∨ b = 0 := by simp [div_eq_mul_inv]
+#align div_eq_zero_iff div_eq_zero_iff
+
+theorem div_ne_zero_iff : a / b ≠ 0 ↔ a ≠ 0 ∧ b ≠ 0 :=
+  div_eq_zero_iff.not.trans not_or
+#align div_ne_zero_iff div_ne_zero_iff
+
+theorem Ring.inverse_eq_inv (a : G₀) : Ring.inverse a = a⁻¹ := by
+  obtain rfl | ha := eq_or_ne a 0
+  · simp
+  · exact Ring.inverse_unit (Units.mk0 a ha)
+#align ring.inverse_eq_inv Ring.inverse_eq_inv
+
+@[simp]
+theorem Ring.inverse_eq_inv' : (Ring.inverse : G₀ → G₀) = Inv.inv :=
+  funext Ring.inverse_eq_inv
+#align ring.inverse_eq_inv' Ring.inverse_eq_inv'
+
+end GroupWithZero
+
+section CommGroupWithZero
+
+-- comm
+variable [CommGroupWithZero G₀] {a b c d : G₀}
+
+-- see Note [lower instance priority]
+instance (priority := 10) CommGroupWithZero.cancelCommMonoidWithZero :
+    CancelCommMonoidWithZero G₀ :=
+  { GroupWithZero.cancelMonoidWithZero,
+    CommGroupWithZero.toCommMonoidWithZero with }
+#align comm_group_with_zero.cancel_comm_monoid_with_zero CommGroupWithZero.cancelCommMonoidWithZero
+
+-- See note [lower instance priority]
+instance (priority := 100) CommGroupWithZero.toDivisionCommMonoid : DivisionCommMonoid G₀ :=
+  { ‹CommGroupWithZero G₀›, GroupWithZero.toDivisionMonoid with }
+#align comm_group_with_zero.to_division_comm_monoid CommGroupWithZero.toDivisionCommMonoid
+
+end CommGroupWithZero
+
+section NoncomputableDefs
+
+open Classical
+
+variable {M : Type _} [Nontrivial M]
+
+/-- Constructs a `GroupWithZero` structure on a `MonoidWithZero`
+  consisting only of units and 0. -/
+noncomputable def groupWithZeroOfIsUnitOrEqZero [hM : MonoidWithZero M]
+    (h : ∀ a : M, IsUnit a ∨ a = 0) : GroupWithZero M :=
+  { hM with
+    inv := fun a => if h0 : a = 0 then 0 else ↑((h a).resolve_right h0).unit⁻¹,
+    inv_zero := dif_pos rfl,
+    mul_inv_cancel := fun a h0 => by
+      change (a * if h0 : a = 0 then 0 else ↑((h a).resolve_right h0).unit⁻¹) = 1
+      rw [dif_neg h0, Units.mul_inv_eq_iff_eq_mul, one_mul, IsUnit.unit_spec],
+    exists_pair_ne := Nontrivial.exists_pair_ne }
+#align group_with_zero_of_is_unit_or_eq_zero groupWithZeroOfIsUnitOrEqZero
+
+/-- Constructs a `CommGroupWithZero` structure on a `CommMonoidWithZero`
+  consisting only of units and 0. -/
+noncomputable def commGroupWithZeroOfIsUnitOrEqZero [hM : CommMonoidWithZero M]
+    (h : ∀ a : M, IsUnit a ∨ a = 0) : CommGroupWithZero M :=
+  { groupWithZeroOfIsUnitOrEqZero h, hM with }
+#align comm_group_with_zero_of_is_unit_or_eq_zero commGroupWithZeroOfIsUnitOrEqZero
+
+end NoncomputableDefs
+
+-- porting note: Seems like `assert_exists` is not implemented yet, so cannot deal with this yet
+/- ./././Mathport/Syntax/Translate/Command.lean:719:14: unsupported user command assert_not_exists
+-/
+-- Guard against import creep
```

Comment:
Can you copy across the command from mathlib3? When Mario next updates the syntax file for mathport this will work; it's just a very new command.

Reply:
mathlib#17416